CodeReview/src/commands/CFocus.cpp

```diff
@@ -61,41 +63,44 @@ bool CFocus::canInterpret(Visualization::Item*, Visualization::Item*,
 Interaction::CommandResult* CFocus::execute(Visualization::Item*, Visualization::Item*,
 				const QStringList&, const std::unique_ptr<Visualization::Cursor>&)
 {
+	focusStep({}, {}, {});
+	return new Interaction::CommandResult{};
+}
 
+bool CFocus::focusStep(Visualization::Item *, QKeySequence, Interaction::ActionRegistry::InputState)
+{
 	Visualization::VisualizationManager::instance().mainScene()->removeOverlayGroup("focusOverlay");
 
-	auto focusInformation = focusList_.value(currentStep_);
+	auto focusInformations = focusList_.values(currentStep_);
 
-	if (focusInformation.type_ == FocusInformation::None){
+	if (focusInformations.isEmpty()){
 		currentStep_ = 0;
-		focusInformation = focusList_.value(currentStep_);
+		focusInformations = focusList_.values(currentStep_);
 	}
 
-	if (focusInformation.type_ == FocusInformation::None)
-		return new Interaction::CommandResult{};
+	if (focusInformations.isEmpty())
+		return false;
+
+	for (auto focusInformation : focusInformations)
+	{
 
-	auto focusItem = CodeReviewManager::instance().overlayForNodeReviews(focusInformation.node_);
+		auto focusItem = CodeReviewManager::instance().overlayForNodeReviews(focusInformation.node_);
 
-	switch (focusInformation.type_) {
-		case FocusInformation::Center:
-			Visualization::VisualizationManager::instance().mainView()->
-					centerOn(focusItem);
-			break;
-		case FocusInformation::Highlight:
+		if (focusInformation.type_.testFlag(FocusInformation::Center))
+		{
+			Visualization::VisualizationManager::instance().mainView()->centerOn(focusItem);
+		}
+
+		if (focusInformation.type_.testFlag(FocusInformation::Highlight))
```

Comment:
At the beginning of the method I call `removeOverlayGroup("focusOverlay")`. This seems to work.